[xinniyee] iP

xinniyee opened the pull request:
### Buddy: Your Personal Task Manager

> “The secret of getting ahead is getting started.” – Mark Twain

Buddy is here to simplify your life by helping you keep track of your tasks effortlessly. 
It's:
- User-friendly with a simple text-based interface.
- Customizable to meet your needs.
- Efficient and ~~FAST~~ SUPER FAST. 🚀

**Features:**

- [ ] Add, delete, and manage tasks with ease.
- [ ] Mark tasks as done or undone.
- [ ] Search for tasks using keywords.

**How to Get Started**
1. Clone the repository from [https://github.com/xinniyee/ip.git](url)
2. Compile and run the program using Java.
3. Start adding and managing your tasks with Buddy!

Here's how the main method looks:

```
public class Main {
    public static void main(String[] args) {
        Buddy.run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

xinniyee committed:
done level 4

xinniyee committed:
Error handling

xinniyee committed:
Added Delete Command

xinniyee committed:
Add Checks for UI testing

xinniyee committed:
Add Storage for tasks

xinniyee committed:
Inteprete Date and Time

xinniyee committed:
Add more OOP

xinniyee committed:
Add Buddy package

xinniyee committed:
Merge remote-tracking branch 'origin/add-gradle-support'

xinniyee committed:
Use Gradle

xinniyee committed:
Add Junit Tests

xinniyee committed:
Add JavaDoc comments

xinniyee committed:
Follow the Coding Standard

xinniyee committed:
Add Find method

xinniyee committed:
Merge branch 'branch-A-CodingStandard'

xinniyee committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/buddy/Parser.java
#	src/main/java/buddy/TaskList.java
#	src/main/java/buddy/Ui.java

K1mcheee commented on src/main/java/buddy/TaskList.java:
```diff
@@ -0,0 +1,139 @@
+    /**
+     * Constructs a TaskList with the provided list of tasks.
+     *
+     * @param tasks the initial list of tasks
```
Perhaps you missed out the "." for param comments, same for the other javadocs below

K1mcheee commented on src/main/java/buddy/Buddy.java:
```diff
@@ -0,0 +1,73 @@
+                exit();
+                break;
+            }
+            Parser.parseCommand(userInput,taskList);
```
I think you may have missed out the space in `(userInput, taskList)`

K1mcheee commented on src/main/java/buddy/Parser.java:
```diff
@@ -0,0 +1,115 @@
+        } else if (input.isEmpty()) {
+            Ui.printError("Please provide an input.");
+        } else if (input.startsWith("todo") || input.startsWith("deadline")
+                || input.startsWith("event") || input.startsWith("find")){
```
Remember it'll be good to add the space between the closing bracket and opening braces :)

K1mcheee commented on src/main/java/buddy/Storage.java:
```diff
@@ -0,0 +1,104 @@
+                String to = parts[4];
+                return new Event(description, from, to, isDone);
+            default:
+                return null; // If the task type is not recognized
```
It'll be good to configure your IDE to not auto-indent the case blocks so it fits the coding standard

javentankangle commented on src/main/java/buddy/Ui.java:
```diff
@@ -0,0 +1,129 @@
+    /**
+     * Prints a line of underscores to separate sections in the output.
+     */
+    public static void printLine() {
```
I like how made use of helper methods

javentankangle commented on src/main/java/buddy/TaskList.java:
```diff
@@ -0,0 +1,139 @@
+     */
+    public void listTasks() {
+        Ui.printTaskList(tasks);
+    }
```
I like how the method are named such that it is easy to understand 👍

javentankangle commented on src/main/java/buddy/TaskList.java:
```diff
@@ -0,0 +1,139 @@
+     */
+    public int size() {
+        return tasks.size();
+    }
```
I didn't like how this method is named. Perhaps there could a verb could be used to make it clearer?

javentankangle commented on src/main/java/buddy/Parser.java:
```diff
@@ -0,0 +1,115 @@
+        }
+    }
+
+    /**
```
Perhaps a switch statement could be used to improve readability?

javentankangle reviewed (commented):
LTGM. A few small changes to be made. Great!

K1mcheee commented on src/main/java/buddy/Buddy.java:
```diff
@@ -0,0 +1,73 @@
+ * Handles initialization, user interaction and execution flow.
+ */
+public class Buddy {
+    private static final String FILE_PATH = Paths.get("data", "Buddy.txt").toString();
```
great job following the naming conventions!

K1mcheee commented on src/main/java/buddy/Storage.java:
```diff
@@ -0,0 +1,104 @@
+package buddy;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import java.util.ArrayList;
+import java.util.List;
```
I like how you grouped the imports!

K1mcheee reviewed (commented):
I liked how you encapsulated your classes! LGTM, just some nits with some spacing.

xinniyee committed:
Add CheckStyle

xinniyee committed:
Change storage method

xinniyee committed:
Implement GUI with JavaFX

xinniyee committed:
Update JavaDocs for the Ui class

- Added detailed documentation for all methods in the Ui class.
- Improved clarity and consistency in method descriptions.
- Explicitly described input parameters and return values for each method.
- Ensured proper formatting and terminology for easier understanding and maintenance.
- Enhanced readability to align with best practices for JavaDocs.

This update ensures that the Ui class is well-documented and ready for use.

xinniyee committed:
Add assert statements to improve error detection and ensure correctness

- Added assert statements to validate task operations like task addition, deletion, and status changes.
- Included checks for invalid task indices to verify correct error handling.
- Ensured that task list operations and interactions are performed as expected during testing.
- Assert statements help catch potential issues early, improving code robustness and reliability.

xinniyee committed:
Improve code quality

xinniyee committed:
Merge pull request #2 from xinniyee/branch-A-Assertions

Add assert statements to improve error detection and ensure correctness

xinniyee committed:
Merge branch 'master' into branch-A-CodeQuality
Merge master into branch-A-CodeQuality to sync with latest changes.

xinniyee committed:
Implement fuzzy search using Levenshtein Distance for typo-tolerant keyword matching

xinniyee committed:
Merge pull request #3 from xinniyee/branch-A-CodeQuality

Improve Code Quality

xinniyee committed:
Handle "/" in command

xinniyee committed:
Improve Welcome Message

xinniyee committed:
Merge branch 'branch-A-Personality'

xinniyee committed:
Correct 'find'

xinniyee committed:
Handle commands with capital letters

xinniyee committed:
Improve Welcome Message

xinniyee committed:
Add Ui.png

xinniyee committed:
Add Error handling

xinniyee committed:
Handle output of Buddy

xinniyee committed:
Add User Guide

xinniyee committed:
Update README

xinniyee committed:
Update README.md

xinniyee committed:
Shorten too long methods

xinniyee committed:
Save local changes to README.md

xinniyee committed:
Save local changes to README.md
